core/unit_test/TestAtomics.hpp

```diff
@@ -440,106 +441,80 @@ T LoopVariantSerial(int loop, int test) {
     case 3: return ExchLoopSerial<T>(loop);
   }
 
+  Kokkos::abort("unreachable");
   return 0;
 }
 
 template <class T, class DeviceType>
-bool Loop(int loop, int test) {
+void Loop(int loop, int test) {
   T res       = LoopVariant<T, DeviceType>(loop, test);
   T resSerial = LoopVariantSerial<T>(loop, test);
 
-  bool passed = true;
-
-  if (resSerial != res) {
-    passed = false;
-
-    std::cout << "Loop<" << typeid(T).name() << ">( test = " << test
-              << " FAILED : " << resSerial << " != " << res << std::endl;
-  }
-
-  return passed;
+  ASSERT_EQ(res, resSerial) << "Loop<" << typeid(T).name() << ">(loop=" << loop
+                            << ",test=" << test << ")";
 }
 
-}  // namespace TestAtomic
-
-namespace Test {
-
 TEST(TEST_CATEGORY, atomics) {
   const int loop_count = 1e4;
 
-  ASSERT_TRUE((TestAtomic::Loop<int, TEST_EXECSPACE>(loop_count, 1)));
-  ASSERT_TRUE((TestAtomic::Loop<int, TEST_EXECSPACE>(loop_count, 2)));
-  ASSERT_TRUE((TestAtomic::Loop<int, TEST_EXECSPACE>(loop_count, 3)));
+  Loop<int, TEST_EXECSPACE>(loop_count, 1);
+  Loop<int, TEST_EXECSPACE>(loop_count, 2);
+  Loop<int, TEST_EXECSPACE>(loop_count, 3);
```

Comment:
Perhaps using an enum instead of `1`, `2`, and `3` can make things a little bit clearer.

Reply:
Leaving it as a potential future refactor work